Kalela/Level-Up
# #158787482 Build cache and dependencies

# What does this PR do?
- Sets up caching of gradle builds to increase build run speeds.
- Separates build.gradle dependencies for easy tracking of their versions and purpose.

### Screenshots
#### Before caching
![image](https://user-images.githubusercontent.com/33759757/42928568-4f8c9164-8b40-11e8-9daf-b341760b9c0b.png)
#### After caching
![image](https://user-images.githubusercontent.com/33759757/42928595-66674708-8b40-11e8-9735-d7fb41f58b04.png)

### How should this be manually tested?
1. Clone the repo.
2. Set up the project with android studio.
3. Go to gradle.properties and remove ```org.gradle.caching=true``` then sync gradle and see the time it takes to run the build.
4. Return ``org.gradle.caching=true``` and run the build twice to see the difference.


### What are the relevant pivotal tracker stories?
158787096
[Manage dependency versions with variables defined in project build.gradle](https://www.pivotaltracker.com/story/show/158787096)
158787482
[Improve gradle build speeds using Gradle Build Cache for Android](https://www.pivotaltracker.com/story/show/158787482)
